server/src/main/java/org/elasticsearch/cluster/routing/IndexRouting.java

```diff
@@ -160,7 +166,11 @@ public void process(IndexRequest indexRequest) {
             // generate id if not already provided
             final String id = indexRequest.id();
             if (id == null) {
-                indexRequest.autoGenerateId();
+                if (creationVersion.onOrAfter(IndexVersions.TIME_BASED_K_ORDERED_DOC_ID_BACKPORT) && indexMode == IndexMode.LOGSDB) {
```

Comment:
This is the difference with the original PR. I had to introduce a different version to account for the fact that in `8.x` branch there is not Lucene 10.